erts/emulator/beam/jit/x86/instr_common.cpp

```diff
@@ -1938,20 +1933,19 @@ void BeamGlobalAssembler::emit_i_test_yield_shared() {
 }
 
 void BeamModuleAssembler::emit_i_test_yield() {
-    Label next = a.newLabel(), entry = a.newLabel();
-
     /* When present, this is guaranteed to be the first instruction after the
      * breakpoint trampoline. */
-    ASSERT(a.offset() % 8 == 0);
+    ASSERT((a.offset() - code.labelOffsetFromBase(currLabel)) ==
+           BEAM_ASM_FUNC_PROLOGUE_SIZE);
 
     emit_enter_frame();
 
-    a.bind(entry);
+    a.lea(ARG3, x86::qword_ptr(currLabel, TEST_YIELD_RETURN_OFFSET));
     a.dec(FCALLS);
-    a.short_().jg(next);
-    a.lea(ARG3, x86::qword_ptr(entry));
-    a.call(yieldEnter);
-    a.bind(next);
+    a.jle(yieldEnter);
```

Comment:
Am I correct in thinking that this will make the process not do a `dec FCALLS` when scheduled back in? Is that a problem?

Reply:
Yes, it'll now match both the ARM JIT and the interpreter. I don't expect it to cause any problems.